cpp/oneapi/dal/algo/knn/test/batch.cpp

```diff
@@ -0,0 +1,339 @@
+
+    auto train_result = this->train(knn_desc, x_train_table, y_train_table);
+    auto infer_result = this->infer(knn_desc, x_infer_table, train_result.get_model());
+
+    this->exact_nearest_indices_check(x_train_table, x_infer_table, infer_result);
+}
+
+KNN_SYNTHETIC_TEST("knn nearest points test random uniform 513x301x17") {
+    SKIP_IF(this->not_available_on_device());
+
+    SKIP_IF(this->is_kd_tree);
+
+    constexpr std::int64_t train_row_count = 513;
+    constexpr std::int64_t infer_row_count = 301;
+    constexpr std::int64_t column_count = 17;
+
+    CAPTURE(train_row_count, infer_row_count, column_count);
+
+    const auto train_dataframe = GENERATE_DATAFRAME(
+        te::dataframe_builder{ train_row_count, column_count }.fill_uniform(-0.2, 0.5));
+    const table x_train_table = train_dataframe.get_table(this->get_homogen_table_id());
+    const auto infer_dataframe = GENERATE_DATAFRAME(
+        te::dataframe_builder{ infer_row_count, column_count }.fill_uniform(-0.3, 1.));
+    const table x_infer_table = infer_dataframe.get_table(this->get_homogen_table_id());
+
+    const table y_train_table = this->arange(train_row_count);
+
+    const auto knn_desc = this->get_descriptor(train_row_count, 1);
+
+    auto train_result = this->train(knn_desc, x_train_table, y_train_table);
+    auto infer_result = this->infer(knn_desc, x_infer_table, train_result.get_model());
+
+    this->exact_nearest_indices_check(x_train_table, x_infer_table, infer_result);
+}
+
+KNN_SYNTHETIC_TEST("knn nearest points test random uniform 16390x20x5") {
+    SKIP_IF(this->not_available_on_device());
+
+    SKIP_IF(this->is_kd_tree);
+
+    constexpr std::int64_t train_row_count = 16390;
+    constexpr std::int64_t infer_row_count = 20;
+    constexpr std::int64_t column_count = 5;
+
+    CAPTURE(train_row_count, infer_row_count, column_count);
+
+    const auto train_dataframe = GENERATE_DATAFRAME(
+        te::dataframe_builder{ train_row_count, column_count }.fill_uniform(-0.2, 0.5));
+    const table x_train_table = train_dataframe.get_table(this->get_homogen_table_id());
+    const auto infer_dataframe = GENERATE_DATAFRAME(
+        te::dataframe_builder{ infer_row_count, column_count }.fill_uniform(-0.3, 1.));
+    const table x_infer_table = infer_dataframe.get_table(this->get_homogen_table_id());
+
+    const table y_train_table = this->arange(train_row_count);
+
+    const auto knn_desc = this->get_descriptor(train_row_count, 1);
+
+    auto train_result = this->train(knn_desc, x_train_table, y_train_table);
+    auto infer_result = this->infer(knn_desc, x_infer_table, train_result.get_model());
+
+    this->exact_nearest_indices_check(x_train_table, x_infer_table, infer_result);
+}
+
+KNN_EXTERNAL_TEST("knn classification hepmass 50kx10k") {
+    SKIP_IF(this->not_available_on_device());
+
+    using Float = double;
+
+    constexpr Float target_score = 0.8;
+
+    constexpr std::int64_t column_count = 28;
```

Comment:
If it works OK, `column_count` should be renamed to `feature_count` or increased by 1.